manager/integration/tests/test_settings.py

```diff
@@ -228,4 +231,98 @@ def test_setting_guaranteed_engine_cpu():
 
     Note: use fixture to restore the setting into the original state
     """
-    pass
+    client = get_longhorn_api_client()
+    core_api = get_core_api_client()
+
+    # Get current guaranteed engine cpu setting and save it to org_setting
+    setting = client.by_id_setting(SETTING_GUARANTEED_ENGINE_CPU)
+    org_setting = setting
+
+    # Get current running instance managers
+    instance_managers = client.list_instance_manager()
+
+    # Set an invalid value, and it should return error
+    try:
+        client.update(setting, value="xxx")
+    except Exception as e:
+        print("Exception when update "
+              "Guaranteed Engine CPU to xxx", e)
+        pass
+
+    # Update guaranteed engine cpu setting to 0.1
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.1",
+                                        "100m", instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 0
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0",
+                                        None, instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 0.2
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.2",
+                                        "200m", instance_managers,
+                                        "Running", True)
+
+    # Update guaranteed engine cpu setting to 10
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "10",
+                                        None, instance_managers,
+                                        "Running", False)
+
+    # Update guaranteed engine cpu setting to 0.25
+    guaranteed_engine_cpu_setting_check(client, core_api, setting, "0.25",
+                                        "250m", instance_managers,
+                                        "Running", True)
+
+    # Create a volume to test
+    vol_name = generate_volume_name()
+    volume = create_and_check_volume(client, vol_name)
+    volume.attach(hostId=get_self_host_id())
+    volume = wait_for_volume_healthy(client, vol_name)
+    assert len(volume.replicas) == 3
+
+    data = write_volume_random_data(volume)
+    check_volume_data(volume, data)
+
+    # Cleanup volume
+    cleanup_volume(client, volume)
+
+    # restore original setting
```

Comment:
Cleanup should be done in the fixture.

Reply:
This is the extra volume I created during the test, I think I need to clean it up, right?

Reply:
The volume should be cleaned up by the `client` fixture automatically.

The setting below should be updated back to the default setting as well.

So those two cleanups won't be necessary.